crates/delta/src/routes/servers/server_edit.rs

```diff
@@ -59,8 +62,16 @@ pub async fn edit(
     }
 
     // Check we are the server owner or privileged if changing sensitive fields
-    if data.owner.is_some() && (user.id != server.owner && !user.privileged) {
-        return Err(create_error!(NotOwner));
+    if data.owner.is_some() {
+        if user.id != server.owner && !user.privileged {
+            return Err(create_error!(NotOwner));
+        }
+
+        if let Totp::Enabled { .. } = account.mfa.totp_token {
```

Comment:
not sure if this check is actually needed, maybe remove it and force every user to confirm it with password or totp?

Reply:
yeah, forcing would be better. if 2FA is off, then this check becomes useless from what I am seeing. check how the login endpoint does it, ig?

Reply:
> check how the login endpoint does it, ig?
not sure where `/auth/session/login` is handled
![image](https://github.com/user-attachments/assets/8ff97ae4-3dd2-43d5-a8b7-474a9eab98fa)

Reply:
oof, i'll take a look and let you know